### ISSUE TITLE
Generating a release on a non default branch GitHub

### ISSUE THREAD

**Issue by tim0git** [go-semantic-release/semantic-release#156]
Hi folks how do i configure the package to generate a release on main when we have set the default branch to be dev?

The commit analyser is reading the default branch. And i would like to to read the main branch.

**Comment by cliedeman** [comment]
Hi @tim0git 

What error are you getting?

The default branch should come directly from the provider

https://github.com/go-semantic-release/provider-github/blob/master/pkg/provider/github.go#L82

**Comment by tim0git** [comment]
Yup. Its finds the default branch correctly. And Analyzes dev. However we are triggering the ci workflow on main. And would like to analyze the commits from the main branch without changing the GitHub default branch from dev to main.

**Comment by tim0git** [comment]
So it does not Error. Howver it does not find any changes.

**Comment by christophwitzko** [comment]
Thanks @cliedeman, you are right. I will leave this issue open for now, until we add support for it.
